test/index.test.ts

```diff
@@ -94,6 +94,21 @@ test('config value that is undefined causes a warning when env is not test', ():
   process.env.APP_ENV = 'test';
 });
 
+test('config value that is undefined does not cause a warning when env is development', (): void => {
+  process.env.APP_ENV = 'development';
+
+  jest.spyOn(global.console, 'warn');
+
+  loadConfig('test/fixtures/validate');
+
+  expect(console.warn).not.toHaveBeenCalledWith('WARNING: Found undefined config value for KEY_3');
```

Comment:
Please change all the `console.warn` instances in this library to `console.log` because warn goes to `stderr`